Dockerfile

```diff
@@ -3,7 +3,8 @@ FROM quay.io/centos/centos:7
 RUN yum install epel-release -y && \
     yum install nginx gettext httpd-tools -y && \
     yum clean all && \
-    mkdir -p /usr/share/nginx/html
+    mkdir -p /usr/share/nginx/html && \
+    mkdir -p /var/cache/nginx
```

Comment:
since we will use this dir for storing cache, it's better to use `VOLUME` instead of in container filesystem, the container should be read only, example [Running nginx in read-only mode](https://hub.docker.com/_/nginx)

Reply:
Good point!

Reply:
made the change you suggested here https://github.com/app-sre/qontract-server/pull/220

Reply:
`/var/run` is for pid file, but our config set it to `/tmp/nginx.pid` [here](https://github.com/app-sre/nginx-gate/blob/2904935ac2283b0e641881a9d3879dc6d32ffdcc/nginx.conf#L4C5-L4C19), also many file writes to `/tmp` at [docker-entrypoint.sh](https://github.com/app-sre/nginx-gate/blob/2904935ac2283b0e641881a9d3879dc6d32ffdcc/docker-entrypoint.sh), so we can mark `/tmp` as volume instead.

Then in Dockerfile, just

```dockerfile
VOLUME /tmp
VOLUME /var/cache/nginx
```

Shouldn't need to `mkdir` or `chmod` anymore, but need to verify it in container and cluster to see if write permission is ok. Can update `docker-compose.yaml` to test locally, and test a deployment in cluster.